- Add runtime dispatch on FreeBSD and OpenBSD
  e91b42110588eb0c3acda53465070afe6c6b471d

Index: CMakeLists.txt
--- CMakeLists.txt.orig
+++ CMakeLists.txt
@@ -511,6 +511,9 @@ include(CheckIncludeFile)
 check_include_file(sys/auxv.h  HAVE_SYS_AUXV_H)
 check_include_file(asm/hwcap.h HAVE_ASM_HWCAP_H)
 
+include(CheckSymbolExists)
+check_symbol_exists(elf_aux_info sys/auxv.h HAVE_ELF_AUX_INFO)
+
 # By default prefer STATIC build (legacy behavior)
 option(BUILD_SHARED_LIBS "Build shared libraries" OFF)
 option(HWY_FORCE_STATIC_LIBS "Ignore BUILD_SHARED_LIBS" OFF)
@@ -540,6 +543,9 @@ if(NOT HAVE_SYS_AUXV_H)
 endif()
 if(NOT HAVE_ASM_HWCAP_H)
   target_compile_definitions(hwy PUBLIC TOOLCHAIN_MISS_ASM_HWCAP_H)
+endif()
+if(HAVE_ELF_AUX_INFO)
+  target_compile_definitions(hwy PUBLIC HWY_HAVE_ELF_AUX_INFO)
 endif()
 target_compile_definitions(hwy PUBLIC "${DLLEXPORT_TO_DEFINE}")
 target_compile_options(hwy PRIVATE ${HWY_FLAGS})
